### ISSUE TITLE
VulnerabilityAnalysisTask (portfolio analysis) may run in multiple threads concurrently, causing high load

### ISSUE THREAD

**Issue by hostalp** [DependencyTrack/dependency-track#1212]
### Current Behavior:
If the task `org.dependencytrack.tasks.VulnerabilityAnalysisTask` doesn't complete before the next run is scheduled to start (every 6 hours it seems) the 2nd one will be started in addition to the already running 1st one. That will make things perform even worse and we have found that in this situation none of these tasks would ever complete (or not within a few days at least).

The situation would repeat at the beginning of another scheduled task run when 3 such tasks would be running concurrently etc. - ultimately consuming all allowed CPU resources.

### Expected Behavior:
If the task is running, start of another instance shouldn't be allowed - just like it's with the task `org.dependencytrack.tasks.MetricsUpdateTask` which can often also take a long time to complete, but it isn't allowed to run more than once at a time. In these cases the following can be seen in logs:
```
INFO [alpine.event.framework.EventService] An singleton event (MetricsUpdateEvent) was received but another singleton event of the same type is already in progress. Skipping.
```

### Environment:
  
  - Dependency-Track Version: 4.3.6
  - Distribution: Docker
  - Database Server: PostgreSQL 12
  - Browser:

**Comment by stevespringett** [comment]
Running concurrent threads is expected behavior. If two projects have uploaded BOMs, they each will kick off their own analysis. If a portfolio analysis is happening, then we would not want to block the analysis of an individual project as that would tie up CI build processes.

I think the proper thing to do is to separate out portfolio analysis from project analysis and restrict only a single occurrence of portfolio analysis.

**Comment by hostalp** [comment]
Yes that's exactly it.

**Comment by hutao4464** [comment]
I also encountered this problem. The track platform ran for a few days, but the bom could not be uploaded.

**Comment by hostalp** [comment]
A little update:
We have developed a small tool that we occassionally use to mass-disable (deactivate) older projects. With that we got from approx. 1300+ active projects to approx. 100 and the processing times dropped as follows:
- MetricsUpdateTask: ~5 hours -> ~7 minutes
- VulnerabilityAnalysisTask ~5 hours -> ~13 minutes

And of course the CPU utilization dropped accordingly.
So the performance doesn't scale very well and it seems that at least the Metrics Update Task performance can be attributed to the Datanucleus library or at least the way it is used as hinted in the comments in #1210

By the way I think that the native DT ability to mass-disable (or even delete) projects based on certain criteria would be generally useful to keep things tidy and in order in the long term.
